posix/unix.c

```diff
@@ -997,21 +1018,24 @@ int unix_poll(unsigned socket, short events)
 	else {
 		if (events & (POLLIN | POLLRDNORM | POLLRDBAND)) {
 			proc_lockSet(&s->lock);
-			if (_cbuffer_avail(&s->buffer) > 0)
+			if (_cbuffer_avail(&s->buffer) > 0 || (s->nonblock && (s->connect != NULL))) {
 				err |= events & (POLLIN | POLLRDNORM | POLLRDBAND);
+			}
 			proc_lockClear(&s->lock);
 		}
 
 		if (events & (POLLOUT | POLLRDNORM | POLLRDBAND)) {
 			if ((conn = unixsock_get_connected(s)) != NULL) {
 				proc_lockSet(&conn->lock);
 				if (conn->type == SOCK_STREAM) {
-					if (_cbuffer_free(&conn->buffer) > 0)
+					if (_cbuffer_free(&conn->buffer) > 0) {
```

Comment:
You should probably add similar change as in above POLLIN case. Poll may be used to wait for non-blocking `connect()` completion (see man page on `connect()` for details).

Reply:
Non-blocking `connect()` completion is handled already by `unixsock_get_connected()` few lines above, so here just checking whether `conn` bufer is ready for writing suffices